src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeFeatureTests.cs

```diff
@@ -33,8 +33,16 @@ public static void DynamicCode()
         [SkipOnMono("IsDynamicCodeCompiled returns false in cases where mono doesn't support these features")]
         public static void DynamicCode_Jit()
         {
-            Assert.True(RuntimeFeature.IsDynamicCodeSupported);
-            Assert.True(RuntimeFeature.IsDynamicCodeCompiled);
+            if (PlatformDetection.IsNativeAot)
+            {
+                Assert.False(RuntimeFeature.IsDynamicCodeSupported);
+                Assert.False(RuntimeFeature.IsDynamicCodeCompiled);
+            }
+            else
+            {
+                Assert.True(RuntimeFeature.IsDynamicCodeSupported);
+                Assert.True(RuntimeFeature.IsDynamicCodeCompiled);
+            }
```

Comment:
Nit: could also be:
```suggestion
            Assert.Equal(!PlatformDetection.IsNativeAot, RuntimeFeature.IsDynamicCodeSupported);
            Assert.Equal(!PlatformDetection.IsNativeAot, RuntimeFeature.IsDynamicCodeCompiled);
```

Reply:
That feels a bit less readable.

This test is currently `SkipOnMono`, but it doesn't have to be (and maybe it shouldn't) - it just needs a way to express the right conditions for Mono and once we do that, the existing structure might be better.

Reply:
Even if I in same camp as you on preferring `if` for readability (actually I would call suggested snnippet harder to parse). Maybe this way would be slightly better. Just want to check on other people if I'm right what's easier.
```
Assert.Equal(PlatformDetection.IsNativeAot, !RuntimeFeature.IsDynamicCodeSupported);
Assert.Equal(PlatformDetection.IsNativeAot, !RuntimeFeature.IsDynamicCodeCompiled);
```

Reply:
That variation gives me the same pause when reading it

Reply:
Interesting. Based on my knowledge people with domain knowledge process separate subdomains as completely separate entities with separate rules. That’s similar to how you use `if`. Let’s see what’s Stephen’s opinion.